Src/FluentAssertions/Events/EventRaisedOrder.cs

```diff
@@ -0,0 +1,20 @@
+using System.Threading;
+
+namespace FluentAssertions.Events
+{
+    /// <summary>
+    /// Tracks the order of events raised by <see cref="FluentAssertions.Events.EventMonitor{T}"/>.
+    /// </summary>
+    internal sealed class EventRaisedOrder
+    {
+        private int orderIndex = -1;
+
+        /// <summary>
+        /// Increments the current order index.
```

Comment:
🤔 There's no concept of an _index_ visible to the outside world. In fact, it generates a unique incrementing sequence in a thread-safe way.

Reply:
Changed to just `sequence` in https://github.com/fluentassertions/fluentassertions/pull/1773/commits/ee0b86ba673e5491c542e9aca797095ad66db2cc.